### PR TITLE
find executorDataMap from declaredFields #551

### PR DESCRIPTION
**Pull request by LuciferYang**
## What changes were proposed in this pull request?
To solve  #551 , change getDeclaredField by name to find target filed from getDeclaredFields.

## How was this patch tested?
exiting test

### CODE REVIEW

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/OapMessageUtils.scala [hunk 1]
```diff
@@ -28,21 +28,24 @@ import org.apache.spark.util.Utils
...
-        val c = Utils.classForName("org.apache.spark.scheduler.cluster.ExecutorData")
-        val executorEndpointField = c.getDeclaredField("executorEndpoint")
-        executorEndpointField.setAccessible(true)
-        val executorEndpoint =
-          executorEndpointField.get(executorData).asInstanceOf[RpcEndpointRef]
-        executorEndpoint.send(message)
-      }
+    // TODO: why we can't just use executorDataMap?
+    val executorDataMapField = classOf[CoarseGrainedSchedulerBackend].getDeclaredFields
+      .find(filed => filed.getName.endsWith("executorDataMap"))
```
Comment (adrian-wang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/OapMessageUtils.scala]: Nit: better break after `(`. when break lines, `(` is preferable than `.`

Comment (LuciferYang) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/OapMessageUtils.scala]: ok~ already extract a lazy value named executorDataMapField~

**Comment by LuciferYang** [comment]
@adrian-wang , In Java, I'm sure it should be called "executorDataMap" and with no prefix, but In scala I don't know which condition let filed name  has prefix. :(

**Comment by adrian-wang** [comment]
shall we merge this to branch-0.3 also?

**Comment by LuciferYang** [comment]
@adrian-wang  ok~

**Comment by adrian-wang** [comment]
merged to master and branch-0.3

**Pull request merged**